File: cpp/0001.two-sum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        vector<int> res;
        map<int, int> value2index;
        map<int, int>::iterator it;
        for (size_t i = 0; i < nums.size(); i++)
        {
            value2index[nums[i]] = i;
        }
        for (size_t i = 0; i < nums.size(); i++)
        {
            it = value2index.find(target - nums[i]);
            if (it != value2index.end())
            {
                if (i == it->second)
                {
                    continue;
                }
                res.push_back(i);
                res.push_back(it->second);
                return res;
            }
        }
    }
};

File: rust/0001_two-sum.rs
impl Solution
{
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32>
    {
        let mut res = Vec::new();
        let n = nums.len();
        'outer: for i in 0..n
        {
            for j in 0..i
            {
                if (nums[i] + nums[j] == target)
                {
                    res.push(j as i32);
                    res.push(i as i32);
                    break 'outer;
                }
            }
        }

        return res;
    }
}
